### ISSUE TITLE
EfficientNetLiteB0Backbone in KerasHub

### ISSUE THREAD

**Issue by cjohn001** [keras-team/keras#20593]
Hello together,
I am currently trying to migrate my code to KerasHub, as it seems for KerasCV the documentation is gone now. In the past I was using the EfficientNetLiteB0Backbone. Seems like it has not been ported to KerasHub. Is there a chance to see this model in KerasHub as well? As I am not seeing it yet I am wondering, is there nowadays maybe a better model to use? I need something small with good classification results for a mobile phone, I think EfficientNetLite was doing the job best with KerasCV.

And in case it will be moved to KerasHub, it would also be great if one could get a pretrained weights for it. Thanks for consideration.

**Comment by edge7** [comment]
Hi, am not part of the Keras team, so this is my personal opinion.

I understand your issue, though. As Keras Hub is still WIP, the Keras CV documentation may still be around. I got a similar issue for an other model recently.
I think EfficientNetLiteB0 is not ported yet, probably it will.
You can still use Keras CV for the time being, look [here](https://github.com/keras-team/keras-cv/tree/master/keras_cv/src/models/backbones/efficientnet_lite) looking at the tests, you might be able to load it correctly even without official documentation.

**Comment by sachinprasadhs** [comment]
Hi, We have a EfficientNet Lite model for the edge devices here https://www.kaggle.com/models/keras/efficientnet/keras/efficientnet_lite0_ra_imagenet which has been ported from `Timm`.
For more details about the model check https://huggingface.co/timm/efficientnet_lite0.ra_in1k

**Comment by cjohn001** [comment]
Hello, thanks for the directions. For the moment this code works for me, I am primary missing pretrained weights and the docs. If it get ported all is fine. For the moment I can live without the docs. Some kind of versioning in the docs would be great.

```
efnetlite_backbone = keras_cv.models.EfficientNetLiteB0Backbone(include_rescaling=True,
                                                                input_shape=(HEIGHT,WIDTH,3))
efnetlite_classifier = keras_cv.models.ImageClassifier(efnetlite_backbone, num_classes=len(class_mapping), activation='sigmoid')
efnetlite_classifier.compile(loss=keras.losses.CategoricalCrossentropy(label_smoothing=0.1),
                   optimizer=keras.optimizers.SGD(momentum=0.9),
                   metrics=["accuracy"])
```

**Comment by cjohn001** [comment]
@sachinprasadhs  I am still trying to figure out how I can get the EfficientNetliteB0Backbone trained to a similar accuracy like was state in the papers of the models. Is there maybe somewhere the training script from ImageNet training available for the model you referenced? Would be a great help to see how the hyperparameters have to be set for training. Thanks for your help!

**Comment by sachinprasadhs** [comment]
@pkgoogle, PTAL

**Comment by pkgoogle** [comment]
Hi @cjohn001,

can you try something like this after installing keras_hub?:
```py
# original timm b0
model = keras_hub.models.ImageClassifier.from_preset("hf://timm/efficientnet_b0.ra_in1k")
# lite variant
model = keras_hub.models.ImageClassifier.from_preset("hf://timm/efficientnet_lite0.ra_in1k")
```

This will load the same preset weights as timm. You can find all the available variants [here](https://github.com/keras-team/keras-hub/blob/master/tools/checkpoint_conversion/convert_efficientnet_checkpoints.py#L51).

**Comment by cjohn001** [comment]
@pkgoogle sorry for the late reply and thanks for the directions. In the meantime I was able to load the weights from here: 

https://github.com/sebastian-sz/efficientnet-lite-keras/releases/tag/v1.0

Unfortunately, it seems like I cannot use keras_hub yet. I want to deploy my models in a mobile application, which I assume requires me to use tensorflow-model-optimization toolkit if I want to use [Sparsity and cluster preserving quantization aware training](https://www.tensorflow.org/model_optimization/guide/combine/pcqat_example).  
Tensorflow-model-optimization toolkit unfortunately forces me to stay on keras2 :( 
Is there maybe another kind of optimization functionality integrated in keras3 already, as a replacement, which I could use instead? 
I would very much like to switch to keras3 as soon as possible, as I think, I could than also directly use timm models with keras3, like described here https://lightning.ai/sitammeur/studios/keras3-with-pytorch-workflow?section=featured. However, However, keras3 is not an option if I cannot prepare the models for deployment. I am quite new to the entire ecosystem, hence  it would be great if you could provide me some directions. Thanks for your help!

**Comment by pkgoogle** [comment]
Hi @cjohn001, If you absolutely need that workflow -- feel free to continue doing so. We usually recommend you get your workflow/system/project working/going before optimization. In which case you have multiple options (including continuing this workflow).

Depending on your use case:

- [mediapipe](https://ai.google.dev/edge/mediapipe/solutions/model_maker). If your usecase fits the tasks listed here -- this is probably the most user friendly way to get started
- [LiteRT](https://github.com/google-ai-edge/litert). If you already have a .tflite model and need to run it on device but you need more control/flexibility than mediapipe allows you can use this library directly
- [AI-Edge-Torch](https://github.com/google-ai-edge/ai-edge-torch). If you need to convert a Pytorch model to a .tflite file to be used either with mediapipe or LiteRT, this is the library you would use. (You can accomplish some of the same optimizations here as well).

Does that answer your question?

**Comment by cjohn001** [comment]
Hello @pkgoogle thanks for the directions. I looked in your mentioned options already. The thought behind my current framework choice was that keras3 gives me full control over the entire build and deployment process and also allows me to build custom solutions based on the building blocks provided with the framework. Switching back to keras2 seems not like a good option to me. Mediapipe seems to me like something nice to get quick results but nothing which supports a full fledged development lifecycle.
However, if I understand you correct, you are saying with Keras3 there is currently no decent way to optimize build models with state of the art techniques like tensorflow-optimization-toolkit for deployment as tfilte? Is there something on the roadmap for Keras3 in this regards?  I believe not having these capabilies will drastically reduce the usability of the framework. With embedded applications in mind would you rather recommend to switch to another framework like pytorch if more control on the build process is required? Thx for your feedback.

**Comment by divyashreepathihalli** [comment]
Hi @cjohn001 
You would be able to load Timm checkpoints for the supported EfficientNetLite architecture directly from HF. So if you have a specific finetuned checkpoint on HF you are interested in you can load that directly.
Here are some of the pre-trained checkpoints available for effecientnet - https://www.kaggle.com/models/keras/efficientnet/keras/

- Keras is a great tuning solution if you are looking into fine-tuning models- here are some guides that can help you with finetuning, custom training loops etc - https://keras.io/guides/
- Quantization is one area we are adding a lot of features in recently - GPTQ has been added and more features are coming
- There is LoRA and QLoRA support out of the box for KerasHub models. 
- We don't support pruning yet.

You can find our roadmap here 
- https://github.com/keras-team/keras/issues/19519
- https://github.com/keras-team/keras-hub/issues/1836

**Comment by cjohn001** [comment]
> Hi [@cjohn001](https://github.com/cjohn001) You would be able to load Timm checkpoints for the supported EfficientNetLite architecture directly from HF. So if you have a specific finetuned checkpoint on HF you are interested in you can load that directly. Here are some of the pre-trained checkpoints available for effecientnet - https://www.kaggle.com/models/keras/efficientnet/keras/
> 
> * Keras is a great tuning solution if you are looking into fine-tuning models- here are some guides that can help you with finetuning, custom training loops etc - https://keras.io/guides/
> * Quantization is one area we are adding a lot of features in recently - GPTQ has been added and more features are coming
> * There is LoRA and QLoRA support out of the box for KerasHub models.
> * We don't support pruning yet.
> 
> You can find our roadmap here
> 
> * [🗺️ Keras Development Roadmap #19519](https://github.com/keras-team/keras/issues/19519)
> * [🗺️ KerasHub Roadmap 🗺️  keras-hub#1836](https://github.com/keras-team/keras-hub/issues/1836)

Hello @divyashreepathihalli, sorry for the late reply and thanks for the directions. I am currently looking into using keras to train a gemma3n model with keras. I am wondering if this is currently possible. I would like to finetune the vision submodel as well as the llm part and deploy it with mediapipe/LiteRTnext on a mobile phone. Is this currently possible as gemma3n is not listed in kerashub. And could you maybe provide directions on how to do this? Mediapipe/ LiteRTnext seems to only support .task files rather than tflite files. Would be great if you could provide a tutorial on how to train and deploy a multimodal model to mobile phones with mediapipe using Keras..
